src/native-automation/request-pipeline/index.ts

```diff
@@ -358,12 +358,18 @@ export default class NativeAutomationRequestPipeline extends NativeAutomationApi
     }
 
     private _getUploadPostData (event: Protocol.Fetch.RequestPausedEvent): string | undefined {
-        if (!event.request.postData)
+        //NOTE: https://github.com/DevExpress/testcafe/issues/8198
+        if (!event.request.postDataEntries || !event.request.postDataEntries.length)
             return void 0;
 
         const contentTypeHeader = event.request.headers['Content-Type'] as string;
-        const postData          = Buffer.from(event.request.postData, 'utf-8');
-        const bodyWithUploads   = injectUpload(contentTypeHeader, postData);
+        const dataBuffers       = [];
+
+        for (const dataEntry of event.request.postDataEntries)
+            dataBuffers.push(Buffer.from(dataEntry.bytes as string, 'base64'));
```

Comment:
Based on the interface, `bytes` can be undefined. In this case, it is better to use the ?? operator and set an empty string, otherwise, it will throw an error if `bytes` is undefined.